lib/scala/runtime-version-manager/src/main/scala/org/enso/runtimeversionmanager/runner/Runner.scala

```diff
@@ -224,18 +224,28 @@ class Runner(
       case None =>
         runtimeVersionManager.withEngineAndRuntime(engineVersion) {
           (engine, runtime) =>
-            NativeExecCommand.apply(
-              engineVersion.toString,
-              engine,
-              logger
-            ) match {
-              case Some(cmd) =>
-                prepareAndRunCommand(engine, cmd)
-              case None =>
-                prepareAndRunCommand(
-                  engine,
-                  JavaExecCommand.forRuntime(runtime)
-                )
+            val ensoLauncher      = Option(System.getenv(Runner.LAUNCHER_ENV_NAME))
+            val requiresJVMRunner = ensoLauncher.exists(_.equals("shell"))
+            if (requiresJVMRunner) {
+              prepareAndRunCommand(
+                engine,
+                JavaExecCommand.forRuntime(runtime)
+              )
+            } else {
+              NativeExecCommand.apply(
+                engineVersion.toString,
+                engine,
+                logger
+              ) match {
+                case Some(cmd) =>
+                  prepareAndRunCommand(engine, cmd)
```

Comment:
As [discovered recently](https://github.com/enso-org/enso/pull/12501#issuecomment-2726213993):

> `ENSO_JVM_OPTS` are used as enso **executable arguments** when launching _native image_ `enso` executable. That means ...  the `ENSO_JVM_OPTS` **change their interpretation** when changing JVM/NI mode.

- what is the meaning of `ENSO_JVM_OPTS` in the proposed `ENSO_LAUNCHER=shell` mode?
- should `ENSO_LAUNCHER` be exposed to end users in next version
- the proper meaning of `ENSO_JVM_OPTS` in non-JVM mode has to be documented

Reply:
OK. Let's have official `--jvm` to switch to the JVM mode. Let's leave the (less prominent) `ENSO_JVM_OPTS` variable to influence options to both NI and JVM modes.